src/cljs/nr/deckbuilder.cljs

```diff
@@ -63,48 +63,50 @@
 (defn lookup
   "Lookup the card title (query) looking at all cards on specified side"
   [side card]
-  (let [q (lower-case (:title card ""))
-        id (:id card)
+  (let [id (:id card)
         cards (filter #(= (:side %) side) @all-cards)
-        exact-matches (filter-exact-title q cards)
         first-id (first (filter #(= id (:code %)) cards))]
-    (cond
-
-      (and id first-id)
+    (if (and id first-id)
       first-id
-
-      (not-empty exact-matches)
-      (take-best-card exact-matches)
-
-      :else
-      (loop [i 2
-             matches cards]
-        (let [subquery (subs q 0 i)]
-          (cond
-            (zero? (count matches))
-            card
-
-            (or (= (count matches) 1) (identical-cards? matches))
-            (take-best-card matches)
-
-            (<= i (count (:title card)))
-            (recur (inc i) (filter-title subquery matches))
-
-            :else
-            card))))))
+      (let [q (lower-case (:title card ""))
+            exact-matches (filter-exact-title q cards)]
+        (if (not-empty exact-matches)
+          (take-best-card exact-matches)
+          (loop [i 2
+                 matches cards]
+            (let [subquery (subs q 0 i)]
+              (cond
+                (zero? (count matches))
+                card
+
+                (or (= (count matches) 1) (identical-cards? matches))
+                (take-best-card matches)
+
+                (<= i (count (:title card)))
+                (recur (inc i) (filter-title subquery matches))
+
+                :else
+                card))))))))
 
 (defn- build-identity-name
   [title setname]
   (if setname
     (str title " (" setname ")")
     title))
 
+(defn lookup-identity
+  "Find an identity card from the card code or title"
+  [side card]
+  (if-let [ident (first (filter #(= (:code card) (:code %)) @identity-cards))]
```

Comment:
Why use `identity-cards` for this instead of just filtering on identities from `all-cards`?